lambda_function.py

```diff
@@ -1,9 +1,161 @@
+
+        self.cloudformation_client = boto3.client("cloudformation")
+        if readonly:
+            self.user_side_stack_cf_filename = "user_side_stack.yaml"
+        else:
+            self.user_side_stack_cf_filename = "user_side_stack.yaml"
+
+    @staticmethod
+    def retrieve_master_account_id() -> str:
+        sts_client = boto3.client('sts')
+        response = sts_client.get_caller_identity()
+        return response.get("Account")
+
+    @staticmethod
+    def generate_external_id() -> str:
+        return str(uuid4())[:8]
+
+    def create_stack_set(self) -> Dict:
+        administrator_role_arn = f"arn:aws:iam::{self.master_account_id}:role/{self.MASTER_ACCOUNT_STACK_SET_ROLE}"
+
+        logger.info(f"Creating StackSet: {self.MASTER_ACCOUNT_PERMISSIONS_STACK_SET_NAME}, "
+                    f"administrator_role_arn : {administrator_role_arn}, "
+                    f"customer_account_external_id: {self.customer_account_external_id}, "
+                    f"customer_account_new_role_name: {self.customer_account_new_role_name}")
+
+        with open(self.user_side_stack_cf_filename) as f:
+            template_body = f.read()
+
+        response = self.cloudformation_client.create_stack_set(
+            StackSetName=Onboarder.MASTER_ACCOUNT_PERMISSIONS_STACK_SET_NAME,
+            Description='Dome9 auto onboarding stack set',
+            TemplateBody=template_body,
+            Parameters=[{'ParameterKey': 'Externalid', 'ParameterValue': "Placeholder"},
+                        {'ParameterKey': 'AccountRoleName', 'ParameterValue': "Placeholder"}],
+            Capabilities=[
+                'CAPABILITY_IAM', 'CAPABILITY_NAMED_IAM', 'CAPABILITY_AUTO_EXPAND'
+            ],
+            AdministrationRoleARN=administrator_role_arn,
+            ExecutionRoleName=Onboarder.CUSTOMER_ACCOUNT_EXECUTION_ROLE_NAME
+        )
+
+        return response
+
+    def create_stack_instances(self) -> None:
+        logger.info(f"Creating stack instance for StackSet: {self.MASTER_ACCOUNT_PERMISSIONS_STACK_SET_NAME}, "
+                    f"region: {self.region_name}, AccountId: {self.customer_account_id}, "
+                    f"NewRoleName {self.customer_account_new_role_name}")
+
+        response = self.cloudformation_client.create_stack_instances(
+            StackSetName=self.MASTER_ACCOUNT_PERMISSIONS_STACK_SET_NAME,
+            Accounts=[self.customer_account_id],
+            Regions=[self.region_name],
+            ParameterOverrides=[{'ParameterKey': 'AccountRoleName', 'ParameterValue': self.customer_account_new_role_name},
+                                {'ParameterKey': 'Externalid', 'ParameterValue': self.customer_account_external_id}],
+            OperationPreferences={
+                'FailureToleranceCount': 0,
+                'MaxConcurrentCount': 3,
+            })
+
+        self.wait_for_stack_operation(response["OperationId"])
+
+    def wait_for_stack_operation(self, operation_id: str) -> None:
```

Comment:
shouldn't raise something if the wait operation exceeds the retries?
add a raise exception at the end of the function (outside the for loop)

Reply:
Yes, all the `waiting` will be redesigned in the next PR.